src/main/java/io/cdap/plugin/http/source/common/pagination/page/JSONUtil.java

```diff
@@ -84,17 +85,18 @@ public static JsonQueryResponse getJsonElementByPath(JsonObject jsonObject, Stri
    * A class which contains information regarding results of searching an element in json by json path.
    */
   public static class JsonQueryResponse {
-    private final String[] retrievedPathParts;
     private final String[] unretrievedPathParts;
+    private final List<String> optionalPathParts;
     private final String retrievedPath;
     private final String unretrievedPath;
     private final JsonElement result;
 
-    public JsonQueryResponse(String[] retrievedPathParts, String[] unretrievedPathParts, JsonElement result) {
-      this.retrievedPathParts = retrievedPathParts;
+    private JsonQueryResponse(String[] retrievedPathParts, String[] unretrievedPathParts,
+                              List<String> optionalPathParts, JsonElement result) {
       this.unretrievedPathParts = unretrievedPathParts;
       this.retrievedPath = "/" + StringUtils.join(retrievedPathParts, '/');
       this.unretrievedPath = "/" + StringUtils.join(unretrievedPathParts, '/');
+      this.optionalPathParts = optionalPathParts;
```

Comment:
Is this field or path? the caller calls this field, while it is called path here.

Reply:
This class represents a parsed JSON, and does it by using some JSON Path (format ``/path/to/field``. However, I am passing option fields from the schema in here, so changed it to ``optionalFields``